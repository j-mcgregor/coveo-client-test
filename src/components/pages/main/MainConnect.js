import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import {
  fetchResults,
  selectProduct
} from '../../../flux/actions/queryActions';
import Main from './MainContainer';

const mapStateToProps = state => ({
  results: state.results
});

const mapDispatchToProps = dispatch => ({
  fetchResults: query => dispatch(fetchResults(query)),
  selectProduct: product => dispatch(selectProduct(product))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Main));
