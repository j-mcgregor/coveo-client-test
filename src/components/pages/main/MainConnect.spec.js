import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import MainConnect from './MainConnect';

const HOCWrapper = (component, props) => (
  <BrowserRouter {...props}>{component}</BrowserRouter>
);

describe('MainConnect', () => {
  it('should render', () => {
    const props = {
      children: []
    };
    const wrapper = mount(HOCWrapper(MainConnect, props));

    expect(wrapper).toMatchSnapshot();
  });
});
