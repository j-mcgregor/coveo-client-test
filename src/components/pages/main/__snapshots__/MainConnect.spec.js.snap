// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainConnect should render with the store 1`] = `ReactWrapper {}`;
